persistence/pg.go

```diff
@@ -241,17 +320,102 @@ func (p *PostgresPersister) Close() error {
 	return p.conn.Close()
 }
 
+func (p *PostgresPersister) checkLockValidity(ctx context.Context, tx *pg.Tx) error {
+	now := time.Now().UTC()
+
+	if now.Before(p.lastUpdate.Add(p.lockUpdateInterval)) {
+		return nil
+	}
+
+	var lock dbInstanceLock
+
+	err := tx.ModelContext(ctx, &lock).Select()
+	if err != nil {
+		return err
+	}
+
+	if lock.LockUpdatedAt != p.lastUpdate {
```

Comment:
I've also added a link to the issue that may contain useful ideas too.

Reply:
This is not safe indeed, one way could be, to have the UUID or the ID of the mux instance in your table.

Then you could do a:
`SELECT. xxxxx  FOR UPDATE SKIP LOCKED` in a TX when you launch a lnmux instance.
This will prevent any other instance to run (if UUID is not nil or if a Tx has selected the row)

In the rare case of a crash that prevents the instance from removing the lock, we could imagine having a field like your `last_update` in the DB, and the instance which has the lock will update (for update skip lock) this field each minute for example.
If other instances (that are waiting for the lock) see that `last_update` is older than a threshold, then one other instance could acquire the lock. 

Imagine the case where an instance didn't update `last_update` but is still alive and another instance acquires the lock. (REALLY UNLIKELY, we don't know why),
You need to check for each DB call if the UUID is still our UUID + a context cancelation that will stop the actual worker/server and make us retry to acquire the lock before continuing.

These are only ideas, I might be wrong

Reply:
In the rare case of a crash that prevents the instance from removing the lock, we could imagine having a field like your `last_update` in the DB, and the instance which has the lock will update (for update skip lock) this field each minute for example.
If other instances (that are waiting for the lock) see that `last_update` is older than a threshold, then one other instance could acquire the lock.

Reply:
Imagine the case where an instance didn't update `last_update` but is still alive and another instance acquires the lock. (REALLY UNLIKELY, we don't know why),
You need to check for each DB call if the UUID is still our UUID + a context cancelation that will stop the actual worker/server and make us retry to acquire the lock before continuing.

These are only ideas, I might be wrong